gpflow/utilities/utilities.py

```diff
@@ -48,6 +49,13 @@ def set_trainable(model: tf.Module, flag: bool):
     """
     Set trainable flag for all `tf.Variable`s and `gpflow.Parameter`s in a module.
     """
+    if not tf.executing_eagerly():
+        tf.print(
+            f"{set_trainable.__name__} may exhibit unexpected behaviour when used outside eager"
+            f" mode",
+            output_stream=logging.warning
+        )
```

Comment:
i spent all morning trying to figure out how to write a unit test for this - it's tricky cos tensorflow checks the `output_stream` value according to some unknown criteria. I can't work out how to satisfy those criteria with a mock, so alas we'll have to live without a test for this

i've checked manually that it works